### PR TITLE
Settings notebook.

### PR DESCRIPTION
**Pull request by markvdw**

### CODE REVIEW

File: doc/source/notebooks/manual/advanced/settings.ipynb [hunk 1]
```diff
@@ -4,11 +4,228 @@
...
+    "m = gpflow.models.SGPR(X, Y, gpflow.kernels.RBF(1), Z=X.copy())\n",
+    "opt.minimize(m)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### WARNING\n",
+    "Because of the static graph compliation of TensorFlow, models defined inside of the context manager will still retain the settins outside the context manager. The zero jitter example above demonstrates this. We define the model inside a context manager with a zero jitter setting. When we try to optimise the model *outside*, the model will use zero jitter, sinde that was the setting's value at the time of model creation. Hence the Cholesky will fail. If we re-define the model with some jitter, the model will optimise just fine."
```
Comment (vdutor) [inline doc/source/notebooks/manual/advanced/settings.ipynb]: compliation -> compilation

Comment (vdutor) [reply doc/source/notebooks/manual/advanced/settings.ipynb]: settins -> settings

Comment (vdutor) [reply doc/source/notebooks/manual/advanced/settings.ipynb]: sinde -> since

File: doc/source/notebooks/manual/advanced/settings.ipynb [hunk 2]
```diff
@@ -4,11 +4,228 @@
...
+    "You can also make `gpflowrc` a hidden file, if you don't want it clutting your home directory, by renaming as `.gpflowrc`.\n",
+    "\n",
+    "Usually, one wants to change the settings for an entire project. We recommend adding a `gpflowrc` file in the working directory to achieve this effect. It is helpful to add assertions / warnings in code to ensure that the correct settings are used. GPflow also allows settings to be adjusted in code, although this comes with some additional difficulties (see below). This is why we recommend using a settings file."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Common uses\n",
```
Comment (vdutor) [inline doc/source/notebooks/manual/advanced/settings.ipynb]: Maybe an extra 'common use' is changing the logging level. Really useful to see which multiple dispatch methods are actually being used.

File: doc/source/notebooks/manual/advanced/settings.ipynb (hunk 1 above)
Comment (markvdw) [reply doc/source/notebooks/manual/advanced/settings.ipynb]: Thanks for catching this.

File: doc/source/notebooks/manual/advanced/settings.ipynb (hunk 2 above)
Comment (markvdw) [reply doc/source/notebooks/manual/advanced/settings.ipynb]: This is now addressed. However, I couldn't get it to work by changing the logging level in code. It only seemed to work if I changed it in `gpflowrc`.

**Pull request merged**